### PR TITLE
rns_shell sub window support

### PR DESCRIPTION
**Pull request by shefayaparwin**
Enabled support to create multiple window ### 

- exposed Created window interface and enhanced it to create window of specified size and type
- Added additional window type : SubWindow for all the secondary windows
- Instead of having global viewbackend. Created view backend per window for support multiple window.
- Fixed : crashing in X11 on closing window
- Fixed : to avoid Blank screen in X11 on window launch, fired expose event notification

### CODE REVIEW

File: rns_shell/platform/graphics/libwpe/WindowLibWPE.cpp [hunk 1]
```diff
@@ -118,7 +128,11 @@ bool WindowLibWPE::initRenderTarget(wpe_view_backend* viewBackend, wpe_renderer_
 #endif
     };
     wpe_renderer_backend_egl_target_set_client(rendererTarget_, &s_client, this);
-    if( viewWidth_ <= 0 || viewHeight_ <= 0) {
+    if(!dimension.isEmpty()) {
+       viewWidth_ =  dimension.width();
+       viewHeight_ = dimension.height();
+    }
+    else if( viewWidth_ <= 0 || viewHeight_ <= 0) {
```
Comment (munezbn) [inline rns_shell/platform/graphics/libwpe/WindowLibWPE.cpp]: Fix else if indentation

File: rns_shell/platform/graphics/x11/WindowX11.cpp [hunk 2]
```diff
@@ -130,7 +133,11 @@ bool WindowX11::initWindow(PlatformDisplay *platformDisplay) {
     Screen *screen = nullptr;
 
     /* Read first screens display resolution*/
-    if(ScreenCount(display) > 0 && (screen = ScreenOfDisplay(display, 0))) {
+    if(!dimension.isEmpty()) {
+       initialWidth =  dimension.width();
+       initialHeight = dimension.height();
+    }
+    else if(ScreenCount(display) > 0 && (screen = ScreenOfDisplay(display, 0))) {
```
Comment (munezbn) [inline rns_shell/platform/graphics/x11/WindowX11.cpp]: Fix else if indentation

**Review by munezbn — APPROVED** [review]

File: rns_shell/platform/graphics/libwpe/PlatformDisplayLibWPE.h [hunk 3]
```diff
@@ -21,15 +21,15 @@ class Display {
 public:
     struct wpe_view_backend* viewBackend() const { return viewBackend_; }
```
Comment (pprabhu-ng) [inline rns_shell/platform/graphics/libwpe/PlatformDisplayLibWPE.h]: do we  need the viewBackend_ in this class now ?

**Review by chatrapathi06 — APPROVED** [review]

File: rns_shell/platform/graphics/libwpe/PlatformDisplayLibWPE.h (hunk 3 above)
Comment (shefayaparwin) [reply rns_shell/platform/graphics/libwpe/PlatformDisplayLibWPE.h]: Good Catch. Not Needed ,it's a dead code now. will remove it.

**Review by pprabhu-ng — APPROVED** [review]

**Pull request merged**